core/server/proxy/src/main/java/alluxio/proxy/s3/signature/utils/AwsAuthV4QueryParserUtils.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.signature.utils;
+
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_DATE;
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_SIGNATURE;
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_CREDENTIAL;
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_SIGNED_HEADER;
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_ALGORITHM;
+import static alluxio.proxy.s3.signature.SignerConstants.X_AMZ_EXPIRES;
+import static alluxio.proxy.s3.signature.SignerConstants.TIME_FORMATTER;
+
+import alluxio.proxy.s3.S3Exception;
+import alluxio.proxy.s3.signature.AwsCredential;
+import alluxio.proxy.s3.signature.SignatureInfo;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+
+/**
+ * Util for getting auth info from query parameters.
+ * <p>
+ * See: https://docs.aws.amazon
+ * .com/AmazonS3/latest/API/sigv4-query-string-auth.html
+ */
+public final class AwsAuthV4QueryParserUtils {
+
+  /**
+   * Function to parse signature.
+   * @param queryParameters
+   * @return SignatureInfo instance
+   * @throws S3Exception
+   */
+  public static SignatureInfo parseSignature(Map<String, String> queryParameters)
+          throws S3Exception {
+    if (!queryParameters.containsKey(X_AMZ_SIGNATURE)) {
+      return null;
+    }
+    validateDateAndExpires(queryParameters);
+    final String rawCredential = queryParameters.get(X_AMZ_CREDENTIAL);
+
+    AwsCredential credential = null;
+    try {
+      credential = AwsCredential.Factory.create(URLDecoder.decode(rawCredential, "UTF-8"));
```

Comment:
Use a reference to the variable in `S3Constants` you make for `"UTF-8"` in this context.

Reply:
do you mean define a variable in S3Constants for "UTF-8"?

Reply:
finish. Instead of using `S3Constants.AUTHORIZATION_CHARSET.name()`.